request-server_test.go

```diff
@@ -345,6 +346,38 @@ func TestRequestFstat(t *testing.T) {
 	checkRequestServerAllocator(t, p)
 }
 
+func TestRequestFsetstat(t *testing.T) {
+	p := clientRequestServerPair(t)
+	defer p.Close()
+	_, err := putTestFile(p.cli, "/foo", "hello")
+	require.NoError(t, err)
+	fp, err := p.cli.OpenFile("/foo", os.O_WRONLY)
+	require.NoError(t, err)
+	err = fp.Truncate(2)
```

Comment:
If I comment out this line, and the `assert.Equal(t, fi.Size(), int64(2))` line, then the test still passes. This is because all of the extra writing to `/foo` is done after a `fp.Seek(int64(2), io.SeekStart)`, which means it will end up as `"hehello"` regardless of whether we’ve done the truncate or not.

But really, we should only be testing `Fsetstat` here, so we should probably just test to make sure that the file after the truncate is `fi.Name() == "foo"`, `fi.Size() == 2` and `body == "he"`.

Maybe followed by a truncate to five, and and a test that `fi.Size() == 5` and `body == []byte{ 'h', 'e', 0, 0 , 0 }`. (Depending on if we want to guarantee behavior as per XSI-standards, or not)